### PR TITLE
feat: OR Query support

### PR DESCRIPTION
**Pull request by MarkDuckworth**
feat: OR Query support

### CODE REVIEW

**Review by ehsannas — COMMENTED** [review]
LGTM. Will take a second look / approve whenever we can actually merge the PR

File: types/firestore.d.ts [hunk 1]
```diff
@@ -2424,6 +2437,127 @@ declare namespace FirebaseFirestore {
...
+  /**
+   * A `CompositeFilter` is used to narrow the set of documents returned
+   * by a Firestore query by performing the logical OR or AND of multiple
+   * {@link Filters}s. `CompositeFilters`s are created by invoking {@link Filter#or}
+   * or {@link Filter#and} and can then be passed to {@link Query#where}
+   * to create a new query instance that also contains the `CompositeFilter`.
+   */
+  export class CompositeFilter extends Filter {
+    private constructor(f);
+  }
```
Comment (ehsannas) [inline types/firestore.d.ts:2565]: Do we need to expose UnaryFilter and CompositeFilter?

File: dev/system-test/firestore.ts [hunk 2]
```diff
@@ -1897,10 +1897,282 @@ describe('Query class', () => {
...
+    );
+
+    // Test with limits (explicit order by DESC): (a==2) || (b == 1) ORDER BY a LIMIT_TO_LAST 1
+    expectDocs(
+      await collection
+        .where(
+          Filter.or(Filter.where('a', '==', 2), Filter.where('b', '==', 1))
+        )
+        .limitToLast(1)
+        .orderBy('a', 'desc')
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply dev/system-test/firestore.ts:2005]: Fixed. Thanks!

File: types/firestore.d.ts (hunk 1 above)
Comment (MarkDuckworth) [reply types/firestore.d.ts]: Removed this from d.ts and API reference docs.

File: types/firestore.d.ts [hunk 3]
```diff
@@ -2424,6 +2437,127 @@ declare namespace FirebaseFirestore {
...
+     * ```
+     */
+    static where(
+      fieldPath: string | firestore.FieldPath,
+      opStr: firestore.WhereFilterOp,
+      value: unknown
+    ): Filter;
+
+    /**
+     * Creates and returns a new [CompositeFilter]{@link CompositeFilter} that performs
```
[the remark this answers is not in the document]
Comment (MarkDuckworth) [reply types/firestore.d.ts]: Fixed. Thanks!

**Review by ehsannas — APPROVED** [review]

**Comment by MarkDuckworth** [comment]
Waiting on backend support. Tests pass against the emulator.

**Comment by ehsannas** [comment]
We probably don't want to auto-merge this? (the CI may have special privileges to prod)

**Review by ehsannas — APPROVED** [review]

**Pull request merged**